packages/neo-one-editor/src/manager/PackageManager.ts

```diff
@@ -67,46 +72,103 @@ export class PackageManager {
     return packages.reduce<{ readonly [name: string]: Module }>((acc, pkg) => ({ ...acc, [pkg.name]: pkg }), {});
   }
 
-  private async fetchPackage(depInfo: DependencyInfo): Promise<Module> {
-    let file;
-    if (depInfo.main !== undefined && depInfo.main.includes('.')) {
-      file = depInfo.main;
-    }
-    const [pkg, pkgJSON, types, subPkgs] = await Promise.all([
-      this.downloadDependency({ name: depInfo.name, version: depInfo.version, file }),
-      this.downloadDependency({ name: depInfo.name, version: depInfo.version, file: PACKAGE_JSON }),
-      depInfo.types === undefined
-        ? Promise.resolve(undefined)
-        : this.downloadDependency({ name: depInfo.name, version: depInfo.version, file: depInfo.types }),
-      this.fetchPackages(depInfo.dependencies),
+  private async fetchPackage({ name, version, types, dependencies }: DependencyInfo): Promise<Module> {
+    const [packageFilePaths, subDependencies] = await Promise.all([
+      this.getAllPackageFiles({ name, version }),
+      this.fetchPackages(dependencies),
+      this.checkTypes({ name, version, types }),
     ]);
 
+    const packageFiles = await Promise.all(
+      packageFilePaths.map(async (filePath) => ({
```

Comment:
This is going to launch hundreds, maybe thousands of requests, which is bad. You need to 
1/ Batch requests using their api
2/ Push them into a queue and only have some N number of fetches running at a time.